site/content/en/latest/tasks/extensibility/extension-server.md

```diff
@@ -0,0 +1,209 @@
+---
+title: "Envoy Gateway Extension Server"
+---
+
+This task explains how to extend Envoy Gateway using an Extension Server. Envoy Gateway
+can be configured to call an external server over gRPC with the xDS configuration _before_ 
+it is sent to Envoy Proxy. The external server can modify the provided configuration 
+programmatically using any semantics supported by the [xDS][] API.
+
+Using an extension server allows vendors to add xDS configuration that Envoy Gateway itself
+doesn't support with a very high level of control over the generated xDS configuration.
+
+**Note:** Modifying the xDS configuration generated by Envoy Gateway may break functionality
+configured by native Envoy Gateway means. Like other cases where the xDS configuration
+is modified outside of Envoy Gateway's control, this is risky and should be tested thoroughly,
+especially when using the same extension server across different Envoy Gateway versions.
+
```

Comment:
not essential, but I would add another note here about Envoy Gateway only supporting a single extension at any given time.